extensions/amp-fx-collection/0.1/providers/amp-fx-presets.js

```diff
@@ -109,4 +109,70 @@ export const Presets = {
       });
     },
   },
+  'fade-in-scroll': {
+    isFxTypeSupported(win) {
+      user().assert(isExperimentOn(win, 'amp-fx-fade-in-scroll'),
+          'amp-fx-fade-in-scroll experiment is not turned on.');
+    },
+    userAsserts(element) {
+      if (!element.hasAttribute('data-margin-start') &&
+        !element.hasAttribute('data-margin-end')) {
+        return;
+      }
+      const marginStart = element.getAttribute('data-margin-start');
+      user().assert(parseFloat(marginStart) >= 0 &&
+        parseFloat(marginStart) < 100, 'data-margin-start must be a number ' +
```

Comment:
Range is [0, 100] interpreted as percentage, correct? The assertion message could disambiguate between [0, 100] and [0..1], since the latter is also a common format for percentages.